pkg/sca/sca.go

```diff
@@ -360,7 +360,9 @@ func generateSharedObjectNameDeps(ctx context.Context, hdl SCAHandle, generated
 			for _, soname := range sonames {
 				libver := sonameLibver(soname)
 
-				if allowedPrefix(path, libDirs) {
+				if allowedPrefix(path, libDirs) &&
+					// .so's in site-packages are not considered provides, they're vendored.
+					!strings.Contains(path, "site-packages") {
```

Comment:
No this is not quite right.

For provides, we should only generate them for public libraries. Meaning `allowdPrefix` check is correct, but is lacking a check for `if isInDir(path, libDirs)` like we do in cmd provides.

As in:
```
# apk info -L systemd | grep -e libudev -e libsystemd
usr/lib/libudev.so.1
usr/lib/libudev.so.1.7.9
usr/lib/systemd/libsystemd-core-256.so
usr/lib/systemd/libsystemd-shared-256.so
```

There should be provides generated for libudev, because it `isInDir` of libDirs.
But usr/lib/systemd/libsystemd* librireis should not generate a public provides, because they are private libraries, not in the library resolution paths of ld-so.

Please change code here to do a check similar to the one done in the cmd provides, as in check for "isInDir"

```
$ lib64/ld-linux-x86-64.so.2 --help
...
Shared library search path:
  (libraries located via /etc/ld.so.cache)
  /lib (system search path)
  /usr/lib (system search path)
  /usr/local/lib (system search path)
  /usr/local/lib64 (system search path)
  /usr/lib (system search path)
  /usr/lib64 (system search path)
  /lib (system search path)
  /lib64 (system search path)
```
But it is not recursive, meaning libraries need to be in those paths directly, not in subdirs.

Reply:
cc @smoser @imjasonh

Reply:
Thanks this is super helpful! I'll update this PR to use that logic, with tests.